source/common/router/header_parser.cc

```diff
@@ -342,28 +355,48 @@ void HeaderParser::evaluateHeaders(Http::HeaderMap& headers,
     headers.remove(header);
   }
 
+  // Temporary storage to hold evaluated values of headers to add and replace. This is required
+  // to execute all formatters using the original received headers.
+  // Only after all the formatters produced the new values of the headers, the headers are set.
+  // absl::InlinedVector is optimized for 4 headers. After that it behaves as normal std::vector.
```

Comment:
nit: absl;;InlineVector is I think capable of any optimal size; it's you that's picking the '4', right?

Can you explain why you picked 4? benchmarks?

Reply:
The reasoning behind picking 4 was that I believe that in the majority of use cases the number of added/replaced headers is rather small. I do not have any concrete data though. I was leaning towards 2,3 or 4. Above that I believe would be over-optimization. The benchmark actually tests adding and replacing 1,2,3,...,10 headers respectively, so it goes beyond 4 and I do not see a significant jump in timing after exceeding 4.

Reply:
sounds good. Throw that in the comment and I'm good to go :)